docs/tutorials/error_analysis.rst

```diff
@@ -0,0 +1,1199 @@
+of the pulsar's orbit: the sign of :math:`\cos( i_\mathrm{p} )` is not known,
+and hence there are two possible :math:`(i_\mathrm{p}, \Omega_\mathrm{p})`
+pairs. The reason for this is that scintillation measurments of a single
+one-dimensional scattering screen can only constrain one component of the
+pulsar's two-dimensional sky-plane velocity (namely, the component in the
+direction set by :math:`\xi`) and there are two possible orbital orientations
+that give the same velocities in that direction.
+
+As a result of the ambiguities, there are two solutions in the spaces of the
+harmonic coefficients and the phenomenological parameters, and there are four
+solutions in physical-parameter space. In this tutorial, we initially show all
+possible solutions, but then zoom in on a single solution to quantify and
+visualize the uncertainties on the parameters at that solution. Here, we pick
+the solution that we will focus on by chosing the sign of the quantity inside
+the absolute-value operation in the model equation and the sign of
+:math:`\cos( i_\mathrm{p} )`:
+
+.. jupyter-execute::
+
+    sol_sign_choice = -1
+    cos_sign_choice = -1
+
+
+The harmonic coefficients
+=========================
+
+This tutorial gives a demonstration of error propagation using two methods:
+
+- Monte Carlo sampling, using Astropy's :py:mod:`~astropy.uncertainty` module.
+- Linear error propagation, using the :py:mod:`uncertainties` package.
+
+For both methods, the input harmonic coefficients first need to be prepared.
+As a sanity check, we will then visualize the fitting results in the parameter
+space in which the fitting was performed (i.e., the space of the harmonic
+coefficients).
+
+
+Monte Carlo sampling
+--------------------
+
+Set the number of samples.
+
+.. jupyter-execute::
+
+    nmc = 40000
+
+Generate random signs to explore both solutions. Specifically, make an Astropy
+:py:class:`~astropy.uncertainty.Distribution` object consisting of +1 and -1
+entries. These set the sign of the quantity inside the absolute-value operation
+in the model equation. Later, we will focus on one of the two solutions by
+selecting one of the signs.
+
+.. jupyter-execute::
+
+    rnd_sign = np.random.randint(low=0, high=2, size=nmc) * 2 - 1
+    sol_sign = unc.Distribution(rnd_sign)
+
+Generate samples of the correlated harmonic coefficients.
+
+.. jupyter-execute::
+
+    hcs = np.random.multivariate_normal(popt, pcov, size=nmc)
+
+    # separate harmonic coefficients
+    hc_es = sol_sign * unc.Distribution(hcs[:, 0] * u.km/u.s/u.pc**0.5)
+    hc_ec = sol_sign * unc.Distribution(hcs[:, 1] * u.km/u.s/u.pc**0.5)
+    hc_ps = sol_sign * unc.Distribution(hcs[:, 2] * u.km/u.s/u.pc**0.5)
+    hc_pc = sol_sign * unc.Distribution(hcs[:, 3] * u.km/u.s/u.pc**0.5)
+    hc_0  = sol_sign * unc.Distribution(hcs[:, 4] * u.km/u.s/u.pc**0.5)
+
+    samp_harc = (hc_es, hc_ec, hc_ps, hc_pc, hc_0)
+
+    samp_harc_all = [dist.distribution.value.tolist() for dist in samp_harc]
```

Comment:
This feels weird to me. Why is the `tolist()` needed? WIthout it, if `np.array` does not work (I think it should), then surely `np.stack` would work, and then you can even give the `axis` so no need for a `.T`

Reply:
Done. Thanks for pointing that out